ivy_tests/test_ivy/test_frontends/test_jax/test_jax_devicearray.py

```diff
@@ -0,0 +1,57 @@
+# global
+import numpy as np
+from hypothesis import given, strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_cmd_line_args
+
+
+@st.composite
+def _reshape_helper(draw):
+    # generate a shape s.t len(shape) > 0
+    shape = draw(helpers.get_shape(min_num_dims=1))
+
+    reshape_shape = draw(helpers.reshape_shapes(shape=shape))
+
+    dtype = draw(helpers.array_dtypes(num_arrays=1))[0]
+    x = draw(helpers.array_values(dtype=dtype, shape=shape))
+
+    is_dim = draw(st.booleans())
+    if is_dim:
+        # generate a permutation of [0, 1, 2, ... len(shape) - 1]
+        permut = draw(st.permutations(list(range(len(shape)))))
+        return x, dtype, reshape_shape, permut
+    else:
+        return x, dtype, reshape_shape, None
+
+
+@handle_cmd_line_args
+@given(
+    x_reshape_permut=_reshape_helper(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.jax.lax.reshape"
+    ),
+)
+def test_jax_reshape_instance_method(
+    x_reshape_permut,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    x, dtype, shape, dimensions = x_reshape_permut
+    helpers.test_frontend_function(
```

Comment:
The testing is still very much wip, Im looking into that today :)

Reply:
we will need to either create a new helper function for instance methods on the frontend arrays or extend `test_frontend_function` to support this. One approach would be to add something like `test_frontend_method` which mirrors `test_method`, and then we create a frontend array instance inside the helper function and then test the specified method. However, it might be easier to simply extend `test_frontend_function` with a flag for converting to frontend array instances. Will have a think and assign a task for this soon.